### PR TITLE
Remove daml-finance from the docs build

### PR DESCRIPTION
**Pull request by brian-weir**
Link to daml finance's github-pages instead.

### CODE REVIEW

File: docs/index/index.rst [hunk 1]
```diff
@@ -17,6 +17,14 @@ Daml Documentation
...
    :caption: Home
 
+.. toctree::
+   :titlesonly:
+   :maxdepth: 0
+   :hidden:
+   :caption: Daml Finance
+
+   Daml Finance <https://digital-asset.github.io/daml-finance/>
+
```
Comment (garyverhaegen-da) [inline docs/index/index.rst:27]: We probably still want a small overview page that includes a link rather than linking directly from the toc.

Comment (brian-weir) [reply docs/index/index.rst:27]: I was hoping for a top level link that didn't need a dropdown (couldn't get that working) and I wasn't even sure where to put this in the TOC. 

I'll add a small overview page for this.

Comment (brian-weir) [reply docs/index/index.rst:27]: Added

**Review by garyverhaegen-da — APPROVED** [review]
Thanks!

**Pull request merged**